numpy/random/_generator.pyx

```diff
@@ -694,21 +694,23 @@ cdef class Generator:
         cdef uint64_t set_size, mask
         cdef uint64_t[::1] hash_set
         # Format and Verify input
+        a_original = a
         a = np.array(a, copy=False)
         if a.ndim == 0:
             try:
                 # __index__ must return an integer by python rules.
                 pop_size = operator.index(a.item())
             except TypeError:
-                raise ValueError("a must an array or an integer")
+                raise ValueError("a must be an array or an integer, "
+                                 f"not {type(a_original)}") from None
             if pop_size <= 0 and np.prod(size) != 0:
                 raise ValueError("a must be a positive integer unless no"
-                                 "samples are taken")
+                                 " samples are taken")
```

Comment:
I think we normally format this with the space at the end of the prevous line

Reply:
Do you recommend using `except TypeError as exc` and then `raise ValueError(...) from exc`, or just leave it as it is in master?

Reply:
Done.  If this is important enough to comment on in a PR, perhaps it is important enough to have in a style guide somewhere?